crates/std_lib/src/hash.nr

```diff
@@ -9,3 +9,22 @@ fn pedersen(_input : [Field]) -> [Field; 2] {}
 
 #[foreign(hash_to_field)]
 fn hash_to_field(_input : [Field]) -> Field {}
+
+// mimc-p/p implementation
+// constants are (publicly generated) random numbers, for instance using keccak as a ROM.
+// You must use constants generated for the native field
+// Rounds number should be ~ log(p)/log(exp)
+// For 254 bit primes, exponent 7 and 91 rounds seems to be recommended
+const ROUNDS: Field = 91;
+fn mimc(x: Field, k: Field, constants: [Field;ROUNDS], exp : Field) -> Field {
+    constrain ROUNDS == 91;     //TODO: to remove once global const is working fine
```

Comment:
Is something broken with global consts when you tried to use it here? Are you referencing PR #350 ? Global consts are currently forced to be constant when they are parsed so they are not polymorphic like other consts.

Reply:
Yes, when I use ROUNDS in the loop: `for i in 1..ROUNDS {..`
I have an error, ROUNDS is a variable and not a const.

Reply:
I thought the global const PR added an implicit `const` to any type annotations of these variables.
You can probably fix this in the meantime by not forcing the variable to be non-const in the type annotation by `const ROUNDS: const Field = 91;` or removing it entirely to `const ROUNDS = 91;`

Reply:
> I thought the global const PR added an implicit `const` to any type annotations of these variables.

It did force an implicit `const` when parsing, that is why I am surprised it error'd out. I want to look into why this is but will have to be a little later in day/week

Reply:
@vezenovm I assume the const issue is not a topic for this PR, can you resolve this conversation if you agree?

Reply:
Yeah either way not an issue for this PR